src/framework/theme/service/style/style.service.ts

```diff
@@ -5,16 +5,38 @@ import {
 } from '../../type';
 import { getThemeValue } from '../theme';
 
-export function createThemedStyle(mapping: ThemedStyleType, theme: ThemeType): StyleType {
-  return Object.keys(mapping).reduce((acc: StyleType, current: string): StyleType => {
-    const mappingValue: any = mapping[current];
+export function createThemedStyle(mapping: ThemedStyleType,
+                                  theme: ThemeType,
+                                  validParams: string[]): StyleType {
 
-    if (mappingValue instanceof Object) {
-      acc[current] = createThemedStyle(mappingValue, theme);
-    } else {
-      acc[current] = getThemeValue(mappingValue, theme, mappingValue);
-    }
+  const reducedMapping: ThemedStyleType = getReducedMapping(mapping, validParams);
+  return Object.keys(reducedMapping).reduce((acc: StyleType, current: string): StyleType => {
+    const mappingValue: any = mapping[current];
+    acc[current] = getThemeValue(mappingValue, theme, mappingValue);
 
     return acc;
   }, {});
 }
+
+export function getReducedMapping(mapping: ThemedStyleType,
```

Comment:
Prefer refactoring `getReducedMapping`, `validate`, and `showWarning` into a single function with more understandable name

It also should be called before `createThemedStyle` so that `createThemedStyle` will only go through mapping keys and apply corresponding theme values

In a few words new function should just filter mapping keys and showing warning if resulting array is not empty

Reply:
Don't think so.

Reply:
We definitely have to remove "extra" params because of react-native validation